### DISCUSSION TITLE
Get boolean for screen size

### DISCUSSION THREAD

**Discussion by ivanjeremic** [chakra-ui/chakra-ui#2221] [Q&A] [answered]
Is there any way to get `true` back if the current breakpoint is active?

```
// First, create an alias for breakpoints
const breakpoints = ["30em", "48em", "62em", "80em"];
// aliases
breakpoints.sm = breakpoints[0];
breakpoints.md = breakpoints[1];
breakpoints.lg = breakpoints[2];
breakpoints.xl = breakpoints[3];
```

**Comment by ljosberinn** [comment, marked as answer]
https://next.chakra-ui.com/docs/hooks/use-breakpoint-value#usage

```js
const breakpoint = useBreakpointValue({ base: "base", sm: "sm", lg: "lg", xl: "xl" })

const isBase = breakpoint === 'sm';
// etc
```

**Reply by ivanjeremic** [reply]
Thanks

**Reply by ivanjeremic** [reply]
@ljosberinn Is there any reason this in on version 1 and not on the newest? I mean the hook `useBreakpointValue`

**Reply by ljosberinn** [reply]
sorry not following, can you rephrase?

**Reply by ivanjeremic** [reply]
I'm not able to import `useBreakpointValue` hook, now I saw this is a separate package on npm called `@chakra-ui/media-query` what is the reason this is separated? thanks

**Reply by ljosberinn** [reply]
this is v1, where you can import it from `@core`, v0.8 doesn't have it

**Reply by ivanjeremic** [reply]
So should I update my core to v1?

**Reply by ljosberinn** [reply]
I'd recommend using v1 if you're starting a project or aren't far in, yes. You can find migration docs in the link from above too: https://next.chakra-ui.com/docs/getting-started

**Comment by ivanjeremic** [comment]
@ljosberinn  I would like to add some conditionals in the props based on the current breakpoint and the methods you showed me and something I found online, but both methods I tried seem to not work correctly always.

```
// Method: 1 >> (This works perfectly)

<Grid
      templateColumns={{
        sm: "1fr",
        md: "1fr",
        lg: "300px 1fr 300px",
        xl: "300px 1fr 300px",
      }}
      templateRows={{
        sm: "20px 1fr 1fr",
        md: "20px 1fr 1fr",
        lg: "100px 1fr 1fr",
        xl: "100px 1fr 1fr",
      }}
      w="100%"
      h="100vh"
>
  boxcontent
</Grid>

// Method: 2 >> (This is buggy, worked the first time now it does not switch the fontSize anymore) <<

const breakpoint = useBreakpointValue({
    sm: "sm",
    md: "md",
    lg: "lg",
    xl: "xl",
  });

  const isSM = breakpoint === "sm";
  const isMD = breakpoint === "md";
  const isLG = breakpoint === "lg";
  const isXL = breakpoint === "xl";

<Heading
    fontSize={ isSM ? "31px" : isMD ? "31px" : isLG ? "31px" : isXL ? "31px" : "" }
>
    Some Heading
</Heading>

// Here the same as above but it seems to work correctly
<Box hidden={isLG ? true : isXL ? true : ""}>
    <BurgerMenu />
</Box>

// This does not work correctly on the small breakboints
<Box
   mb={{ sm: "2em", md: "2em", lg: "2em", xl: "2em" }}
   pt={{ sm: "4em", md: "4em", lg: "4em", xl: "4em" }}
   pl={{ sm: "2em", md: "2em", lg: "2em", xl: "2em" }}
>
 MyBox
</Box>
```

So what would you generally recommend, is the way I try to do this in v1 the correct way? Thanks.

**Reply by ljosberinn** [reply]
re method 2: you always pass `31px` so... what should it update? 🤔 

```js
// Here the same as above but it seems to work correctly
<Box hidden={isLG ? true : isXL ? true : ""}>
    <BurgerMenu />
</Box>
```

that's the same as
```js
<Box hidden={isLG || isXL}>
```

and can you clarify what `does not work correcly on small breakpoints` means? A sandbox would be nice.

**Reply by ivanjeremic** [reply]
> re method 2: you always pass `31px` so... what should it update? 🤔
> 
> ```js
> // Here the same as above but it seems to work correctly
> <Box hidden={isLG ? true : isXL ? true : ""}>
>     <BurgerMenu />
> </Box>
> ```
> 
> that's the same as
> 
> ```js
> <Box hidden={isLG || isXL}>
> ```
> 
> and can you clarify what `does not work correcly on small breakpoints` means? A sandbox would be nice.

This is only a typo in the example I sent here.

Yes I will try to prepare a CodeSandbox

**Reply by ivanjeremic** [reply]
Somewhat hard to prepare a CodeSanbox but I will try but it seems using something like ```size={
          isSM
            ? "80px"
            : isMD
            ? "80px"
            : isLG
            ? "200px"
            : isXL
            ? "1080px"
            : "1080px"
        }``` Is something ChakraUI does not like and does not understand but it is valid JavaScript

**Reply by ljosberinn** [reply]
You're generally missing the `base` breakpoint, which is anything below `sm`. Chakra doesn't do anything special with your JS. If it's valid JS, then it will work.

Here's a sandbox with your code and it works just fine. https://codesandbox.io/s/headless-thunder-4l51i?file=/src/App.js
